vms/platformvm/state/state.go

```diff
@@ -808,6 +837,42 @@ func (s *state) SetSubnetOwner(subnetID ids.ID, owner fx.Owner) {
 	s.subnetOwners[subnetID] = owner
 }
 
+func (s *state) GetSubnetManager(subnetID ids.ID) (ids.ID, []byte, error) {
+	if chainIDAndAddr, exists := s.subnetManagers[subnetID]; exists {
+		return chainIDAndAddr.ChainID, chainIDAndAddr.Addr, nil
+	}
+
+	if chainIDAndAddr, cached := s.subnetManagerCache.Get(subnetID); cached {
+		return chainIDAndAddr.ChainID, chainIDAndAddr.Addr, nil
+	}
+
+	chainIDAndAddrBytes, err := s.subnetManagerDB.Get(subnetID[:])
+	if err == database.ErrNotFound {
+		s.SetSubnetManager(subnetID, ids.Empty, []byte{})
+		return ids.Empty, []byte{}, nil
```

Comment:
Returning `nil` here seems unexpected to me... Why are these values returned?

Reply:
If we call `GetSubnetManager` on existing Subnets, we'll get an `ErrNotFound`. It's not that these are errors, it's that these Subnets don't have managers. Returning `{ids.Empty, []byte{}}` is the empty Subnet manager which is expected until `SetSubnetValidatorManagerTx` is called.

Reply:
I think that this is an unusual place to specify that "No manager is treated as `{ids.Empty, []byte{}}`" is defined in the state package rather than the execution logic.

Reply:
Fair- will handle in a followup PR for the execution logic